clang-10 warning: definition of implicit copy constructor is deprecated

In boost/fusion/view/filter_view/filter_view.hpp:
```
boost/fusion/view/filter_view/filter_view.hpp:62:22: error: definition of implicit copy constructor for 'filter_view<const boost::fusion::tuple<boost::filesystem::path &, boost::process::detail::arg_setter_<char, false> &, boost::process::basic_environment<char> &>, boost::process::detail::is_initializer<mpl_::arg<-1> > >' is deprecated because it has a user-declared copy assignment operator [-Werror,-Wdeprecated-copy]
        filter_view& operator= (filter_view const&);
```

In boost/fusion/container/vector/vector_iterator.hpp:
```
boost/fusion/container/vector/vector_iterator.hpp:48:26: error: definition of implicit copy constructor for 'vector_iterator<boost::fusion::tuple<boost::process::detail::posix::exe_cmd_init<char>, boost::process::detail::posix::env_init<char> >, 0>' is deprecated because it has a user-declared copy assignment operator [-Werror,-Wdeprecated-copy]
        vector_iterator& operator= (vector_iterator const&);
```

In boost/fusion/view/filter_view/filter_view_iterator.hpp:
```
boost/fusion/view/filter_view/filter_view_iterator.hpp:66:26: error: definition of implicit copy constructor for 'filter_iterator<boost::fusion::forward_traversal_tag, boost::fusion::vector_iterator<const boost::fusion::tuple<boost::filesystem::path &, boost::process::detail::arg_setter_<char, false> &, boost::process::basic_environment<char> &>, 0>, boost::fusion::vector_iterator<const boost::fusion::tuple<boost::filesystem::path &, boost::process::detail::arg_setter_<char, false> &, boost::process::basic_environment<char> &>, 3>, boost::process::detail::is_initializer<mpl_::arg<-1> > >' is deprecated because it has a user-declared copy assignment operator [-Werror,-Wdeprecated-copy]
        filter_iterator& operator= (filter_iterator const&);
```

--- a/include/boost/fusion/container/vector/vector_iterator.hpp
+++ b/include/boost/fusion/container/vector/vector_iterator.hpp
@@ -41,6 +41,10 @@ namespace boost { namespace fusion
         vector_iterator(Vector& in_vec)
             : vec(in_vec) {}
 
+        BOOST_CONSTEXPR BOOST_FUSION_GPU_ENABLED
+        vector_iterator(vector_iterator const& rhs)
+            : vec(rhs.vec) {}
+
         Vector& vec;
 
     private:

--- a/include/boost/fusion/view/filter_view/filter_view.hpp
+++ b/include/boost/fusion/view/filter_view/filter_view.hpp
@@ -52,6 +52,11 @@ namespace boost { namespace fusion
         {}
 
         BOOST_CONSTEXPR BOOST_FUSION_GPU_ENABLED
+        filter_view(filter_view const& rhs)
+            : seq(rhs.seq)
+        {}
+
+        BOOST_CONSTEXPR BOOST_FUSION_GPU_ENABLED
         first_type first() const { return fusion::begin(seq); }
         BOOST_CONSTEXPR BOOST_FUSION_GPU_ENABLED
         last_type last() const { return fusion::end(seq); }

--- a/include/boost/fusion/view/filter_view/filter_view_iterator.hpp
+++ b/include/boost/fusion/view/filter_view/filter_view_iterator.hpp
@@ -59,6 +59,10 @@ namespace boost { namespace fusion
         filter_iterator(First const& in_first)
             : first(filter::iter_call(first_converter::call(in_first))) {}
 
+        BOOST_CONSTEXPR BOOST_FUSION_GPU_ENABLED
+        filter_iterator(filter_iterator const& rhs)
+            : first(rhs.first) {}
+
         first_type first;
 
     private:

